pkg/credentials/service_account_credentials_test.go

```diff
@@ -167,7 +167,7 @@ func TestS3CredentialBuilder(t *testing.T) {
 		g.Expect(c.Create(context.TODO(), existingServiceAccount)).NotTo(gomega.HaveOccurred())
 		g.Expect(c.Create(context.TODO(), existingS3Secret)).NotTo(gomega.HaveOccurred())
 
-		err := builder.CreateSecretVolumeAndEnv(scenario.serviceAccount.Namespace, scenario.serviceAccount.Name,
+		err := builder.CreateSecretVolumeAndEnv(scenario.serviceAccount.Namespace, nil, scenario.serviceAccount.Name,
```

Comment:
Do we have a test case with both `annotations` and `storageSecretName` to make sure `annotations` takes precedence?

Reply:
It would be nice to have a new test case because the default config comes with both `annotations` and `storageSecretName`.